sdk/storage/inc/datalake/datalake_options.hpp

```diff
@@ -0,0 +1,943 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/shared_request_options.hpp"
+#include "protocol/datalake_rest_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  /**
+   * @brief Service client options used to initalize DataLakeServiceClient.
+   */
+  struct ServiceClientOptions
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  };
+
+  /**
+   * @brief File system client options used to initalize DataLakeFileSystemClient.
+   */
+  struct FileSystemClientOptions
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  };
+
+  /**
+   * @brief Path client options used to initalize DataLakePathClient.
+   */
+  struct PathClientOptions
+  {
+    std::vector<std::unique_ptr<Azure::Core::Http::HttpPolicy>> policies;
+  };
+
+  /**
+   * @brief Optional parameters for DataLakeServiceClient::ListFilesSystems
+   */
+  struct ListFileSystemsOptions : public SharedRequestOptions
```

Comment:
do we need to add the ` : public ` ?   That's the default for structs right?

Reply:
I still prefer writing it out explicitly.